netann/common_test.go

```diff
@@ -0,0 +1,198 @@
+		startTime: time.Now(),
+		signer:    &mockSigner{}, // returns a nil signature
+		expErr:    errors.New("cannot decode empty signature"),
+	},
+}
+
+// TestUpdateDisableFlag checks the behavior of UpdateDisableFlag, asserting
+// that the proper channel flags are set, the timestamp always increases
+// monotonically, and that the correct errors are returned in the event that the
+// signer is unable to produce a signature.
+func TestUpdateDisableFlag(t *testing.T) {
+	t.Parallel()
+
+	for _, test := range updateDisableTests {
+		tc := test
+		t.Run(test.name, func(t *testing.T) {
+			t.Parallel()
+
+			// Create the initial update, the only fields we are
+			// concerned with in this test are the timestamp and the
+			// channel flags.
+			ogUpdate := &lnwire.ChannelUpdate{
+				Timestamp: uint32(tc.startTime.Unix()),
+			}
+			if !tc.startEnabled {
+				ogUpdate.ChannelFlags |= lnwire.ChanUpdateDisabled
+			}
+
+			// Create new update to sign using the same fields as
+			// the original. UpdateDisableFlag will mutate the
+			// passed channel update, so we keep the old one to test
+			// against.
+			newUpdate := &lnwire.ChannelUpdate{
+				Timestamp:    ogUpdate.Timestamp,
+				ChannelFlags: ogUpdate.ChannelFlags,
+			}
+
+			// Attempt to update and sign the new update, specifying
+			// disabled or enabled as prescribed in the test case.
+			newUpdate2, err := netann.UpdateDisableFlag(
+				tc.signer, pubKey, newUpdate, tc.disable,
+			)
+
+			var fail bool
+			switch {
+
+			// Both nil, pass.
+			case tc.expErr == nil && err == nil:
+
+			// Both non-nil, compare error strings since some
+			// methods don't return concrete error types.
+			case tc.expErr != nil && err != nil:
+				if err.Error() != tc.expErr.Error() {
+					fail = true
+				}
+
+			// Otherwise, one is nil and one is non-nil.
+			default:
+				fail = true
+			}
+
+			if fail {
+				t.Fatalf("expected error: %v, got %v",
+					tc.expErr, err)
+			}
+
+			// Exit early if the test expected a failure.
+			if tc.expErr != nil {
+				return
+			}
+
+			// Verify that returned update has the same pointer as
+			// the one give to UpdateDisableFlag.
+			if newUpdate != newUpdate2 {
+				t.Fatalf("update disable should modify the " +
+					"passed update")
+			}
+
+			// Verify that the timestamp has increased from the
+			// original update.
+			if newUpdate.Timestamp <= ogUpdate.Timestamp {
+				t.Fatalf("update timestamp should be "+
+					"monotonically increasing, "+
+					"original: %d, new %d",
+					ogUpdate.Timestamp, newUpdate.Timestamp)
+			}
+
+			// Verify that the disabled flag is properly set.
+			disabled := newUpdate.ChannelFlags&
+				lnwire.ChanUpdateDisabled != 0
+			if disabled != tc.disable {
+				t.Fatalf("expected disable:%v, found:%v",
+					tc.disable, disabled)
+			}
+
+			// Finally, validate the signature using the router's
+			// verification logic.
+			err = routing.ValidateChannelUpdateAnn(
+				pubKey, 0, newUpdate,
```

Comment:
We'll actually need the capacity here in case the channel supports the `MaxHTLC` flag, otherwise this new update will fail validation.